gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/CompactionJobConfigurator.java

```diff
@@ -249,6 +250,11 @@ protected boolean configureInputAndOutputPaths(Job job, FileSystemDataset datase
     CompactionPathParser.CompactionParserResult rst = parser.parse(dataset);
     this.mrOutputPath = concatPaths(mrOutputBase, rst.getDatasetName(), rst.getDstSubDir(), rst.getTimeString());
 
+    if(this.state.contains(ConfigurationKeys.USE_DATASET_LOCAL_WORK_DIR)) {
+      mrOutputBase = this.state.getProp(MRCompactor.COMPACTION_DEST_DIR);
+      String tmpDir = ".temp";
```

Comment:
Can we put this on ConfigurationKeys as a default, and possibly use this with the other configurations `STAGING_DIR_DEFAULT_SUFFIX`, `OUTPUT_DIR_DEFAULT_SUFFIX`

Reply:
done